bookwyrm/views/annual_summary.py

```diff
@@ -0,0 +1,79 @@
+from django.db.models import Case, When
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+from django.views import View
+
+from bookwyrm import models
+
+
+class AnnualSummary(View):
+    """display a summary of the year"""
+
+    def get(self, request, year):
+        """get response"""
+
+        user = request.user
+        read_shelf = get_object_or_404(user.shelf_set, identifier="read")
+        read_shelf_books_in_year = (
+            models.ShelfBook.objects.filter(shelf=read_shelf)
+            .filter(user=user)
+            .filter(shelved_date__year=year)
+            .order_by("shelved_date", "created_date", "updated_date")
+        )
+        read_book_ids_in_year = [i.book.id for i in read_shelf_books_in_year]
+        preserved = Case(
+            *[When(pk=pk, then=pos) for pos, pk in enumerate(read_book_ids_in_year)]
+        )
+        read_books_in_year = models.Edition.objects.filter(
+            id__in=read_book_ids_in_year
+        ).order_by(preserved)
+
+        """pages stats queries"""
+        read_books_with_pages = read_books_in_year.filter(pages__gte=0).order_by(
+            "pages"
+        )
+        pages_list = [p.pages for p in read_books_with_pages]
+
+        pages_total = 0
+        pages_average = 0
+        book_pages_lowest = 0
+        book_pages_highest = 0
+        if len(pages_list) > 0:
+            pages_total = sum(pages_list)
```

Comment:
You can get these page stats using the queryset, rather than using a list:
```python3
from django.db.models import Max, Min, Avg, Sum
stats = read_books_with_pages.aggregate(Max("pages"), Sum("pages"), Min("pages"), Avg("pages"))
```

And you'll get an object with the stats in this format:
```
{'pages__max': 3990, 'pages__min': 1, 'pages__sum': 129623, 'pages__avg': 451.64808362369337}
```